src/status_im/ui/screens/profile/contact/views.cljs

```diff
@@ -55,51 +62,97 @@
+                                [:show-popover {:view :share-chat-key :address public-key}])}
+                   [icons/icon :main-icons/share styles/contact-profile-detail-share-icon]]]}])
+
+(defn profile-details-list-view [contact]
+  [list/flat-list {:data                    [contact]
+                   :default-separator?      true
+                   :key-fn                  :public-key
+                   :render-fn               render-detail}])
 
-(defn profile-info [{:keys [public-key]}]
+(defn profile-details [contact]
   [react/view
-   [profile-info-contact-code-item public-key]])
+   [list-header/list-header (i18n/label :t/profile-details)]
+   [profile-details-list-view contact]])
 
-(defn block-contact-action [{:keys [blocked? public-key]}]
+(defn block-contact-action [{:keys [blocked? public-key] :as contact}]
   [list/render-action
    {:label               (if blocked?
                            (i18n/label :t/unblock-contact)
                            (i18n/label :t/block-contact))
-    :icon                :main-icons/cancel
     :action              (if blocked?
                            #(re-frame/dispatch [:contact.ui/unblock-contact-pressed public-key])
-                           #(re-frame/dispatch [:contact.ui/block-contact-pressed public-key]))
+                           #(re-frame/dispatch [:bottom-sheet/show-sheet
+                                                {:content        sheets/block-contact
+                                                 :content-height 160}
+                                                contact]))
     :accessibility-label (if blocked?
                            :unblock-contact
                            :block-contact)}
    {:action-style           styles/block-action
     :action-label-style     styles/block-action-label
     :icon-opts              styles/block-action-icon-opts}])
 
-(defview profile []
-  (letsubs [contact [:contacts/current-contact]]
-    [react/view profile.components.styles/profile
-     [status-bar/status-bar]
-     [profile-contact-toolbar]
-     [react/scroll-view
-      [react/view profile.components.styles/profile-form
-       [profile.components/profile-header
-        {:contact              contact
-         :editing?             false
-         :allow-icon-change?   false}]]
-      [list/action-list (actions contact)
-       {:container-style        styles/action-container
-        :action-style           styles/action
-        :action-label-style     styles/action-label
-        :action-subtext-style   styles/action-subtext
-        :action-separator-style styles/action-separator
-        :icon-opts              styles/action-icon-opts}]
-      [react/view {:style {:height 16}}]
-      [block-contact-action contact]
-      [react/view styles/contact-profile-info-container
-       [profile-info contact]]]]))
+(defn- header-in-toolbar [{:keys [photo-path] :as account}]
+  (let [displayed-name (multiaccounts/displayed-name account)]
```

Comment:
this should be done in the subscription

Reply:
@yenda does it apply here https://github.com/status-im/status-react/blob/09c8d858ed087dc6402f989559d8b17fe0f7cd8a/src/status_im/ui/screens/profile/user/views.cljs#L79 as well?

Reply:
yes we try to avoid doing any computations in the views

Reply:
@yenda any suggestion on how put it in subscription?

i tryed 

```
  (views/letsubs [list-ref (reagent/atom nil)
                  contact  [:contacts/current-contact]
                  displayed-name       (multiaccounts/displayed-name contact)]
```

but i get "unknown".

haven't found a good example in the code...

Reply:
Leave it as it was for now it's fine